control-plane/Dockerfile

```
@@ -0,0 +1,136 @@
+
+# dev copies the binary from a local build
+# -----------------------------------
+FROM hashicorp/consul-k8s-control-plane:latest as dev
+
+# BIN_NAME is a requirement in the hashicorp docker github action 
+ARG BIN_NAME=consul-k8s-control-plane
+COPY pkg/bin/linux_amd64/${BIN_NAME} /bin
+
+
+# ===================================
+# 
+#   Release images.
+#
+# ===================================
+
+
+# default release image
+# -----------------------------------
+# This Dockerfile creates a production release image for the project. This
+# downloads the release from releases.hashicorp.com and therefore requires that
+# the release is published before building the Docker image.
+#
+# We don't rebuild the software because we want the exact checksums and
+# binary signatures to match the software and our builds aren't fully
+# reproducible currently.
+FROM alpine:3.15 AS release-default
+
+# NAME and VERSION are the name of the software in releases.hashicorp.com
+# and the version to download. Example: NAME=consul VERSION=1.2.3.
+ARG BIN_NAME=consul-k8s
+ARG VERSION
+
+LABEL name=${BIN_NAME} \
+      maintainer="Team Consul Kubernetes <team-consul-kubernetes@hashicorp.com>" \
+      vendor="HashiCorp" \
+      version=${VERSION} \
+      release=${VERSION} \
+      summary="consul-k8s-control-plane provides first-class integrations between Consul and Kubernetes." \
+      description="consul-k8s-control-plane provides first-class integrations between Consul and Kubernetes."
+
+# Set ARGs as ENV so that they can be used in ENTRYPOINT/CMD
+ENV BIN_NAME=${BIN_NAME}
+ENV VERSION=${VERSION}
+
+
+# TARGETOS and TARGETARCH are set automatically when --platform is provided.
+ARG TARGETOS TARGETARCH
+
+# This is the location of the releases.
+ENV HASHICORP_RELEASES=https://releases.hashicorp.com
+
+# Create a non-root user to run the software.
+RUN addgroup ${BIN_NAME} && \
+    adduser -S -G ${BIN_NAME} 100
+COPY dist/${TARGETOS}/${TARGETARCH}/${BIN_NAME} /bin/
+
+USER 100
+CMD /bin/${BIN_NAME}
+
+# alternate release image, just for the sake of example. In this case we're using
+# debian as the base image just to make the image different from the default alpine one.
+#
+# The use cases for alternate images are things like defining an additional UBI compatible
+# image, or an image with a different function than the main image, e.g. Waypoint's ODR images.
+# -----------------------------------
+# Dockerfile for consul-k8s with UBI as its base image. Used for running on
+# OpenShift.
+#
+# This Dockerfile creates a production release image for the project. This
+# downloads the release from releases.hashicorp.com and therefore requires that
+# the release is published before building the Docker image.
+#
+# We don't rebuild the software because we want the exact checksums and
+# binary signatures to match the software and our builds aren't fully
+# reproducible currently.
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.5 as release-alternate
```

Could we call it `ubi`?
```suggestion
FROM registry.access.redhat.com/ubi8/ubi-minimal:8.5 as ubi
```

---

Sure!